### PR TITLE
fix(DTFS2-6822): fix linting in typescript services; remove typescript support from webpack builds

### PR DESCRIPTION
**Pull request by lmarrai-sw**
## Introduction

Since we introduced Typescript support to the repos, we've had a few issues with linting

## Resolution

I have:
- installed `@typescript-eslint/eslint-plugin` and `@typescript-eslint/eslint-parser` in the root package.json and configured these to be used by all of the (non-Azure) services for TS files _only_
  - also upgraded these packages in `external-api` where they are also imported
- configured all of the (non-Azure) services to extend the `plugin:@typescript-eslint/recommended-type-checked` config 
  - except for `external-api` because it had too many linting errors with this config to fix in this PR. I will create a separate ticket to add this to `external-api`.
- changed how the `azure-env/index.ts` file is used (the function now runs when the file is imported, instead of having to import the function and run it) to avoid linting errors asking for all other imports to run before the function is called
- installed `@types/express` in all (non-Azure) packages that did not already have it
- removed support for Typescript files in the `scripts` directories of `portal`, `gef-ui`, and `tfm-ui`, because `npm run build` was generating very different output with the TS support and we haven't been able to test that the new output works correctly. I'll raise a separate ticket for adding TS support to the webpack builds
- replaced the `optimize-css-assets-webpack` webpack plugin in `tfm-ui` with the `css-minimizer-webpack-plugin`
  - I had to do this because the `optimize-css-assets-webpack` plugin does not support webpack v5 and we use webpack v5 in tfm-ui: so `npm` would not let me install new packages in `tfm-ui` since it could not resolve the dependencies with these 2 conflicting restrictions
- removed some packages from `tfm-ui` that haven't been used since we removed the GraphQL endpoints from `tfm-api`
- installed `tslib` in any of the TS services that didn't have it, since our TS output assumes that package is available

### CODE REVIEW

File: external-api/.eslintrc.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ module.exports = {
     'eslint:recommended',
     'plugin:import/recommended',
     'plugin:import/typescript',
+    // TODO DTFS2-6822: Raise a ticket to increase this to 'plugin:@typescript-eslint/recommended-type-checked'
```
Comment (lmarrai-sw) [inline external-api/.eslintrc.js:8]: Flagging TODO

File: portal/webpack.common.config.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ const CssMinimizerPlugin = require('css-minimizer-webpack-plugin');
 
 module.exports = {
   entry: {
-    main: './scripts/main.ts',
+    main: './scripts/main.js',
```
Comment (lmarrai-sw) [inline portal/webpack.common.config.js:7]: I should raise a ticket to add TS support to the webpack files

File: gef-ui/package-lock.json [hunk 3]
```diff
@@ -12326,127 +12478,6 @@
...
-      },
-      "engines": {
-        "node": ">=8"
-      }
-    },
-    "node_modules/ts-loader/node_modules/yallist": {
-      "version": "4.0.0",
-      "resolved": "https://registry.npmjs.org/yallist/-/yallist-4.0.0.tgz",
-      "integrity": "sha512-3wdGidZyq5PB084XLES5TpOSRA3wjXAlIWMhum2kRcv/41Sn2emQ0dycQW4uZXLejwKvg6EsvbdlVL+FYEct7A=="
-    },
```
Comment (oscar-richardson-softwire) [inline gef-ui/package-lock.json:12449]: Why were all of these removed?

File: gef-ui/.eslintrc.js [hunk 4]
```diff
@@ -1,3 +1,7 @@
+const baseParserOptions = {
+  ecmaVersion: 2020,
+};
+
```
Comment (oscar-richardson-softwire) [inline gef-ui/.eslintrc.js:4]: Nice and DRY, like it 👍

File: portal-api/tsconfig.eslint.json [hunk 5]
```diff
@@ -3,7 +3,7 @@
     "include": [
         "src",
         "api-tests",
-        "_mocks_",
+        "__mocks__",
```
Comment (oscar-richardson-softwire) [inline portal-api/tsconfig.eslint.json:6]: Good catch 👍

File: portal/package-lock.json [hunk 6]
```diff
@@ -3589,15 +3588,6 @@
...
     },
-    "node_modules/ajv-keywords": {
-      "version": "3.5.2",
-      "resolved": "https://registry.npmjs.org/ajv-keywords/-/ajv-keywords-3.5.2.tgz",
-      "integrity": "sha512-5p6WTN0DdTGVQk6VjcEju19IgaHudalcfabD7yhDGeA6bcQnmL+CpveLJq/3hvfwd1aof6L386Ougkx6RfyMIQ==",
-      "dev": true,
-      "peerDependencies": {
-        "ajv": "^6.9.1"
-      }
-    },
```
Comment (oscar-richardson-softwire) [inline portal/package-lock.json:3600]: Why was this removed?

File: portal/package-lock.json [hunk 7]
```diff
@@ -12888,20 +12746,6 @@
...
-      "dev": true,
-      "peer": true,
-      "bin": {
-        "tsc": "bin/tsc",
-        "tsserver": "bin/tsserver"
-      },
-      "engines": {
-        "node": ">=14.17"
-      }
-    },
```
Comment (oscar-richardson-softwire) [inline portal/package-lock.json:12904]: Why was this removed?

File: trade-finance-manager-api/tsconfig.eslint.json [hunk 8]
```diff
@@ -1,11 +1,8 @@
 {
     "extends": "./tsconfig.json",
     "include": [
-        "server",
+        "src",
         "api-tests",
-        "component-tests",
-        "scripts",
-        "test-helpers",
```
Comment (oscar-richardson-softwire) [inline trade-finance-manager-api/tsconfig.eslint.json:8]: Good catch, this directories don't exist so I'm guessing this tsconfig was originally copied across from another service and not changed or something.

File: gef-ui/package-lock.json [hunk 9]
```diff
@@ -46,7 +46,6 @@
         "path": "0.12.7",
         "redis": "^3.1.2",
         "sanitize-html": "^2.11.0",
-        "ts-loader": "^9.5.0",
```
[the remark this answers is not in the document]
Comment (lmarrai-sw) [reply gef-ui/package-lock.json:49]: Yep that's exactly right

File: trade-finance-manager-api/tsconfig.eslint.json (hunk 8 above)
Comment (lmarrai-sw) [reply trade-finance-manager-api/tsconfig.eslint.json:8]: Yep I think exactly that

File: portal/package-lock.json (hunk 6 above)
Comment (lmarrai-sw) [reply portal/package-lock.json:3600]: It's hard to say on this one, since it was done by `npm` automatically

File: portal/webpack.common.config.js (hunk 2 above)
Comment (lmarrai-sw) [reply portal/webpack.common.config.js:7]: https://ukef-dtfs.atlassian.net/browse/DTFS2-6831
